File: src/main/java/me/lulu/hypixelutilities/resource/ResourceGroup.java
package me.lulu.hypixelutilities.resource;

import com.google.common.collect.Lists;
import net.minecraft.init.Blocks;
import net.minecraft.item.Item;
import net.minecraft.item.ItemStack;

import java.util.ArrayList;
import java.util.List;

public class ResourceGroup implements Resource {

    private String name;
    private List<Resource> resources;

    public ResourceGroup(String name, Resource... resources) {
        this.name = name;
        this.resources = Lists.newArrayList(resources);
    }

    @Override
    public boolean hasEnough() {

        for (Resource resource : resources) {
            if (resource.hasEnough())
                return true;
        }

        return false;
    }

    @Override
    public String getName() {
        return name;
    }

    @Override
    public List<ItemStack> getRequire() {
        List<ItemStack> all = new ArrayList<>();

        for (Resource resource : resources) {
            all.addAll(resource.getRequire());
        }

        return all;
    }


}

File: src/main/java/me/lulu/hypixelutilities/resource/ResourceGroup.kt
package me.lulu.hypixelutilities.resource

import com.google.common.collect.Lists
import net.minecraft.item.ItemStack
import java.util.*

class ResourceGroup(override val name: String, vararg resources: Resource?) : Resource {
    private val resources: List<Resource>
    override fun hasEnough(): Boolean {
        for (resource in resources) {
            if (resource.hasEnough()) return true
        }
        return false
    }

    override val require: List<ItemStack>
        get() {
            val all: MutableList<ItemStack> = ArrayList()
            for (resource in resources) {
                all.addAll(resource.require)
            }
            return all
        }

    init {
        this.resources = Lists.newArrayList<Resource>(*resources)
    }
}
